gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * (c) Copyright 2016 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.gradle.dist.service;
+
+import com.palantir.gradle.dist.asset.AssetDistributionPlugin;
+import com.palantir.gradle.dist.pod.PodDistributionPlugin;
+import com.palantir.gradle.dist.service.tasks.CopyLauncherBinariesTask;
+import com.palantir.gradle.dist.service.tasks.CopyYourkitAgentTask;
+import com.palantir.gradle.dist.service.tasks.CopyYourkitLicenseTask;
+import com.palantir.gradle.dist.service.tasks.CreateCheckScriptTask;
+import com.palantir.gradle.dist.service.tasks.CreateInitScriptTask;
+import com.palantir.gradle.dist.service.tasks.DistTarTask;
+import com.palantir.gradle.dist.service.tasks.LaunchConfigTask;
+import com.palantir.gradle.dist.tasks.ConfigTarTask;
+import com.palantir.gradle.dist.tasks.CreateManifestTask;
+import java.io.File;
+import java.util.stream.Collectors;
+import org.gradle.api.InvalidUserCodeException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.JavaExec;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.application.CreateStartScripts;
+import org.gradle.api.tasks.bundling.Compression;
+import org.gradle.api.tasks.bundling.Jar;
+import org.gradle.api.tasks.bundling.Tar;
+import org.gradle.util.GFileUtils;
+
+public final class JavaServiceDistributionPlugin implements Plugin<Project> {
+    private static final String GO_JAVA_LAUNCHER_BINARIES = "goJavaLauncherBinaries";
+    private static final String GO_JAVA_LAUNCHER = "com.palantir.launching:go-java-launcher:1.5.1";
+    private static final String GO_INIT = "com.palantir.launching:go-init:1.5.1";
+    public static final String GROUP_NAME = "Distribution";
+    private static final String SLS_CONFIGURATION_NAME = "sls";
+
+    @SuppressWarnings("checkstyle:methodlength")
+    public void apply(Project project) {
+        if (project.getPlugins().hasPlugin(AssetDistributionPlugin.class)) {
```

Comment:
this should be withPlugin since this will only ever throw if this plugin is applied after the asset distribution plugin

Reply:
I think this is fine since that would imply that this only runs after the 2nd plugin got applied.
And it most likely will fail to register a duplicate extension before it even gets to that, which means you get less good errors.